pkg/attestation/crafter/runners/githubaction.go

```diff
@@ -16,16 +16,27 @@
 package runners
 
 import (
+	"context"
 	"fmt"
 	"os"
 
 	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/pkg/attestation/crafter/runners/oidc"
 )
 
-type GitHubAction struct{}
+type GitHubAction struct {
+	oidcClient oidc.Client
+}
 
 func NewGithubAction() *GitHubAction {
-	return &GitHubAction{}
+	client, err := oidc.NewOIDCGitHubClient(context.Background())
+	if err != nil {
+		client = oidc.NewNoOPClient()
```

Comment:
add a comment on why we need to do this. Maybe we should add an additional property in runner context called `runner_authenticated` that we can use to identify if the runner was authenticated using oidc or not (fallback)